src/stackdriver-nozzle/nozzle/nozzle.go

```diff
@@ -20,17 +21,48 @@ func (e *PostMetricError) Error() string {
 }
 
 type Nozzle struct {
-	LogHandler    Sink
-	MetricHandler Sink
-	Heartbeater   heartbeat.Heartbeater
+	LogSink    Sink
+	MetricSink Sink
+
+	Heartbeater heartbeat.Heartbeater
+
+	done chan struct{}
+}
+
+func (n *Nozzle) Start(fhClient firehose.Client) (errs chan error, fhErrs <-chan error) {
+	n.Heartbeater.Start()
+	n.done = make(chan struct{})
+
+	errs = make(chan error)
+	messages, fhErrs := fhClient.Connect()
+	go func() {
+		for {
+			select {
+			case envelope := <-messages:
+				err := n.handleEvent(envelope)
+				if err != nil {
+					errs <- err
+				}
+			case <-n.done:
+				return
+			}
+		}
+	}()
+
+	return errs, fhErrs
+}
+
+func (n *Nozzle) Stop() {
+	n.Heartbeater.Stop()
+	n.done <- struct{}{}
```

Comment:
The way I see the done channel is as receiving a "signal" rather than a value. Sending false to the done channel does not make much sense which is why I am leaning towards an empty struct instead.